staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -77,8 +77,6 @@ type Event struct {
 	// AuditLevel at which event was generated
 	Level Level
 
-	// Time the request reached the apiserver.
-	Timestamp metav1.Time
```

Comment:
> not mutate CreationTimestamp after the object is created

I think I've just joined this effort and therefore don't have a preference towards existing solution, that's why I'm not sure it's a right approach. The fact that audit code creates event object once and then mutates it is an implementation detail, it could've been implemented by creating a new object on each stage. Moreover, IMO creating a new object per stage seems like a more intuitive approach from outside of the system. I don't think an implementation detail should influence the API here

> it is one object for us developers

It is not for the API developers but for the developers that will be writing the consumer service, so I would argue that it has to be intuitive for them, not for us

We can do it both ways, with the `StageTimestamp` field or with the `RequestStartedTimestamp` field, the question is what approach will be the most intuitive and will cause the least problems in the future

@sttts WDYT? If you also have a preference towards the `StageTimestamp` field, then I'm OK with it :)